### DISCUSSION TITLE
Ideas for funding

### DISCUSSION THREAD

**Discussion by programmerjake** [obi1kenobi/cargo-semver-checks#715] [General]
I think you may want to apply for a [NLnet](https://nlnet.nl/) grant for working on some non-trivial task(s), e.g. for adding external dependency support. Grants can be up to €50000 until you've completed one, then up to €250000. Grants pay out based on completed tasks rather than hours worked. you just have to demonstrate that your grant provides some benefit to the EU public, which is easy in this case since people in the EU benefit from cargo-semver-checks.

**Comment by obi1kenobi** [comment]
Thanks, I appreciate the suggestion — I'm planning to apply!

If there are any other good avenues for open-source funding, I'd love to hear about them too! Bonus points if you or someone you personally know has had experience applying already and can share what the process was like.

**Comment by programmerjake** [comment]
I've been part of Libre-SOC for >5yr, we've been funded by NLnet for most of that so me and one other member have been able to work on it full time (and quite a few others in their spare time) until a few months ago (serious people problems recently arose that weren't NLnet's fault, hence why I'm hesitant to bring up my experience). In my experience, applying for and getting grants is relatively easy, you submit an initial application, then once the current 2mo application period ends, if NLnet likes it they will contact you with any questions they have, it's essential to respond quickly so you don't get put at the back of the queue. after that you have to pass an external audit (if the grant is small enough, this step may be skipped). after that, you come up with the list of tasks and how much money to allocate to them (they should be relatively coarse so you don't run into as many issues if plans change, though they should be fine enough so you can complete them often enough to get paid semi-regularly), then you will submit them to NLnet and they and you will sign a memorandum of understanding at which point the grant is active and you can get paid for tasks you completed by submitting a request for payment through their web interface they'll give you a link to, that link is like a password so must not be shared publicly.

**Reply by obi1kenobi** [reply]
That sounds great! I've applied for the April 1 cycle, and I'm going to keep my fingers crossed 🤞

**Comment by programmerjake** [comment]
i will also mention that you can split the payout for a task among multiple people in ratios of your choosing. each person can request payment for multiple tasks at a time so they (NLnet and you) don't have a huge amount of overhead from a bunch of €100 payments
